### DISCUSSION TITLE
New-Comers To Linux

### DISCUSSION THREAD

**Discussion by Zardon703** [oxen-io/lokinet#2205] [Ideas]
Good day, just 2 questions;
1. Why not separate the 'curl' command from the 'echo' command in the gui installation side of the website.
2. Why not specify that '$(lsb_release -sc)' means the version of (lets say deb) the distro that is being used. 

The reason I'm recommending it is that it's confusing to new comers (such as myself) who was struggling for 6 months for such a simple problem

Thanks for taking the time to read my minnie rant. '-')/

**Comment by jagerman** [comment]
> Why not separate the 'curl' command from the 'echo' command in the gui installation side of the website.

They are on separate lines now; did you mean there should be blank lines between them?

> Why not specify that '$(lsb_release -sc)' means the version of (lets say deb) the distro that is being used.

`$(lsb_release -sc)` is a shortcut that allows, in most cases, to be able to copy-and-paste the command and have it work on your local system, but there are cases where it doesn't work (such as a barebones base install).  There is more detail about the distro release value to put there below, in the `Long is good, will read:` section.

**Comment by Zardon703** [comment]
> They are on separate lines now; did you mean there should be blank lines between them?
 
 On the cli side there is a difference, but on the gui side it looks like 1 command

**Reply by jagerman** [reply]
Thanks for the feedback; I've added empty lines between the commands for clarity.

**Comment by Zardon703** [comment]
> `$(lsb_release -sc)` is a shortcut that allows, in most cases, to be able to copy-and-paste the command and have it work on your local system, but there are cases where it doesn't work (such as a barebones base install). There is more detail about the distro release value to put there below, in the `Long is good, will read:` section.

 True, but when you're running a distro like Parrot OS, the `$(lsb_release -sc)` command will see it as *ara* instead of *bullseye* and say "lokinet not found" when trying to install (which was my main issue)

**Reply by rjc3rd** [reply]
- focal    (Ubuntu 20.04)
    - jammy    (Ubuntu 22.04)
    - lunar    (Ubuntu 23.04)

Linux Mint 21.2 is based on Ubuntu 22.04 "jammy".
echo "deb [arch=amd64] https://deb.oxen.io/ jammy main" | sudo tee /etc/apt/sources.list.d/oxen.list

Just find out what version of Ubuntu your disto is based on and change $(lsb_release) with the codename for your Ubuntu version.

Possibly Lokinet devs could just include a list of Ubuntu codenames
instead of using the $(lsb_release) since this only work for Ubuntu
in their repository, even though their repository works for probably all
Ubuntu derivatives anyway if you just change it to the Ubuntu codename.

I understand why they used $(lsb_release) to simplify the instructions, but
it breaks all other Ubuntu based installs. Unless they just add other codenames
to their repository for most popular Ubuntu based distros.

Also, I had to add [arch=amd64] for my processor architecture.
I think the Lokinet devs should add this to the instructions as well.
Or at least mention it to users who get error about architecture 
while running sudo apt update or sudo apt install lokinet commands.

@jagerman
